SalesforceSDK/Analytics/Store/EventStoreManager.cs

```diff
@@ -0,0 +1,224 @@
+        {
+            get { return _filenameSuffix; }
+            set { _filenameSuffix = value; }
+        } 
+
+        public string EncryptionKey
+        {
+            get { return _encryptionKey; }
+            set { _encryptionKey = value; }
+        }
+
+        public EventStoreManager(string fileNameSuffix, string encryptionKey)
+        {
+            FilenameSuffix = fileNameSuffix;
+            EncryptionKey = encryptionKey;
+            _rootDir = FileSystem.Current.LocalStorage;
+        }
+
+        public async Task StoreEventAsync(InstrumentationEvent instrumentationEvent)
+        {
+            if (instrumentationEvent == null || string.IsNullOrEmpty(instrumentationEvent.ToJson().ToString()))
+            {
+                LoggingService.Log("Invalid Event", LoggingLevel.Error);
+                return;
+            }
+            //TODO: Add check for shouldstoreevent
+            var fileName = instrumentationEvent.EventId + _filenameSuffix;
+            //Open file
+            IFile file = await _rootDir.CreateFileAsync(fileName, CreationCollisionOption.OpenIfExists);
+            //Wrtie to file after encrypting contents
+            //TODO: Enrypt the contents
+            await file.WriteAllTextAsync(instrumentationEvent.ToJson().ToString());
+        }
+
+        public async Task StoreEventsAsync(List<InstrumentationEvent> instrumentationEvents)
+        {
+            if (instrumentationEvents == null || instrumentationEvents.Count == 0)
+            {
+                LoggingService.Log("No events to store", LoggingLevel.Error);
+                return;
+            }
+            if (!await ShouldStoreEvent())
+            {
+                return;
+            }
+            foreach (var instrumentationEvent in instrumentationEvents)
+            {
+                await StoreEventAsync(instrumentationEvent);
+            }
+        }
+
+        public async Task<InstrumentationEvent> FetchEventAsync(string eventId)
+        {
+            if (string.IsNullOrEmpty(eventId))
+            {
+                LoggingService.Log("Invalid event ID supplied: " + eventId, LoggingLevel.Error);
+                return null;
+            }
+            var fileName = eventId + _filenameSuffix;
+            var file = await _rootDir.GetFileAsync(fileName);
+            return await FetchEventAsync(file);
+        }
+
+        public async Task<List<InstrumentationEvent>> FetchAllEventsAsync()
+        {
+            var files = await _rootDir.GetFilesAsync();
+            var events = new List<InstrumentationEvent>();
+
+            foreach (var file in files)
+            {
+                var instrumentationEvent = await FetchEventAsync(file.Name);
+                if (instrumentationEvent != null)
+                {
+                    events.Add(instrumentationEvent);
+                }
+            }
+
+            return events;
+        }
+
+        public async Task<bool> DeleteEventAsync(string eventId)
```

Comment:
Are the even Ids a set list?  Or can they be any string?  If they are a set list then maybe make an enum list with the allowed events?  If not then ignore this comment 🐹